### PR TITLE
Clear exchange input after an order has been made

### PR DESCRIPTION
**Pull request by 4855062503**
In reference to https://github.com/zap-me/ops/issues/191

### CODE REVIEW

**Comment by zap-janitor** [comment]
🚀 Deployed on https://630d1e573e4a701f8a22251e--bitforge-test.netlify.app

**Review by djpnewton — CHANGES_REQUESTED** [review]
It would be more reliable to move these changes to just after we have succesfully created an order.

ie. just before we show the order screen

```Navigator.push(
            context,
            MaterialPageRoute(
                builder: (context) => OrderScreen(order, widget.websocket)))
```

**Comment by 4855062503** [comment]
I considered that, but thought what if the user wanted to go back and edit the order by a small amount, and thought they may not want to rewrite the order entirely if the input was cleared.

But if you think it's better that way I'll change it.

**Comment by djpnewton** [comment]
> I considered that, but thought what if the user wanted to go back and edit the order by a small amount, and thought they may not want to rewrite the order entirely if the input was cleared.
> 
> But if you think it's better that way I'll change it.

it only clears **after** an order is created . we are talking about in the `_exchange()` function once the "create order" button is clicked

File: lib/exchange.dart [hunk 1]
```diff
@@ -387,6 +387,11 @@ class _ExchangeWidgetState extends State<ExchangeWidget> {
     showAlertDialog(context, 'creating order..');
     var res = await beOrderCreate(_market.symbol, _side, _amount);
     Navigator.pop(context);
+    _amountController.text = '';
+    _receiveController.text = '';
+    setState(() {
+      _validAmount = false;
+    });
     res.when(
         (order) => Navigator.push(
```
Comment (djpnewton) [inline lib/exchange.dart:396]: you arent checking if the call was successful.

it needs to go in *this* block.

ie. **only** when the order was created successfully.

**Review by djpnewton — CHANGES_REQUESTED** [review]

**Comment by djpnewton** [comment]
need squash

**Comment by 4855062503** [comment]
@djpnewton 
Squashed.

**Review by djpnewton — APPROVED** [review]

**Pull request merged**